src/Stomp/Network/Observer/ConnectionObserver.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/*
+ * This file is part of the Stomp package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Stomp\Network\Observer;
+
+
+use Stomp\Transport\Frame;
+
+/**
+ * Interface ConnectionObserver defines the observable events on a connection.
+ *
+ * @package Stomp\Network\Observer
+ */
+interface ConnectionObserver
+{
+    /**
+     * Indicates that during a read call no frame was received, but an EOL line.
+     *
+     * @return void
+     */
+    public function emptyLineReceived();
+
+    /**
+     * Indicates that during a read call no frame was received, but an EOL line.
+     *
+     * @return void
+     */
+    public function emptyBuffer();
+
+    /**
+     * Indicates that a frame has been received.
+     *
+     * @param Frame $frame that has been received
+     * @return void
+     */
+    public function receivedFrame(Frame $frame);
+
+    /**
+     * Indicates that a frame has been transmitted.
```

Comment:
the word `transmit` does not imply a direction, does it?

`received` fits as it describes a incoming frame, but transmitted isnt a good fit IMO.

maybe `sentFrame` or somthing?